dds/DCPS/DataReaderImpl.cpp

```diff
@@ -3282,6 +3287,54 @@ DataReaderImpl::get_ice_endpoint()
   return TransportClient::get_ice_endpoint();
 }
 
+DDS::ReturnCode_t DataReaderImpl::setup_deserialization()
+{
+  const DDS::DataRepresentationIdSeq repIds =
+    get_effective_data_rep_qos(qos_.representation.value);
+  bool success = false;
+  if (cdr_encapsulation()) {
+    Encoding::Kind encoding_kind;
```

Comment:
could this object's scope be reduced (to inside the for loop)?

Reply:
Sure.